addons/mail/static/tests/chat_window_test.js

```diff
@@ -0,0 +1,46 @@
+odoo.define('mail.chat_window_test', function (require) {
+"use strict";
+
+var testUtils = require('web.test_utils');
+var Widget = require('web.Widget');
+var ChatWindow = require('mail.ExtendedChatWindow');
+
+QUnit.module('mail', {}, function () {
+
+    QUnit.test('chat_window: close using ESCAPE key', function (assert) {
+    assert.expect(2);
+
+    function createParent(params) {
+        var widget = new Widget();
+
+        testUtils.addMockEnvironment(widget, params);
+        return widget;
+    }
+    var messages = [{
+        attachment_ids: [],
+        author_id: ["1", "John Doe"],
+        body: "A message",
+        date: moment("2016-12-20 09:35:40"),
+        displayed_author: "John Doe",
+        id: 1,
+        is_note: false,
+        is_starred: false,
+        model: 'partner',
+        res_id: 2
+    }];
+    var parent = createParent({
+        data: {},
+    });
+
+    var chat_window = new ChatWindow(parent, 1, "user", false, messages, {});
```

Comment:
also, we cannot test that the window is closed at the end, but we can test that the close_chat_session event has been triggered. You can add these lines:

    chatWindow.on('close_chat_session', null, function () {
        assert.ok(true, "chat window should trigger a close event");
    });

(and you will need to change the assert.expect accordingly)